### PR TITLE
Backend service

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
New backend reference app has been added.

general idea is that the `apps/ae_backend_service/lib/backend_session.ex` is where the backend implements it's logic. The interactive client (follow the readme to get it going) now allows backend to spin up channels and work will be dispatched to the `backend_session.ex` which currently just naively signs everything.

channels are identified and persisted using their `channel_id`. A returning client (initiator) needs to provide the ´channel_id´ in the backend endpoint [(webserver)](apps/ae_channel_interface/lib/ae_channel_interface_web/controllers/user_controller.ex) The same `channel_id` is needed for the client to connect to it's fsm. No `channel_id` indicates a fresh new channel is requested.

When the client (initiator) requests a new (rest/get) channel using `http://127.0.0.1:4000/connect/new?client_account=ak_SVQ9RvinB2E8pio2kxtZqhRDwHEsmDAdQCQUhQHki5QyPxtMh&port=1610` the response contains the assumed client configuration which must be respected.

### CODE REVIEW

**Comment by AleksandarFilipov** [comment]
Removed WIP to that this can be reviewed and merged. Further modifications requires updated node which is not available on master yet. https://github.com/aeternity/aeternity/pull/3170

Next branch on top of this is expected to be https://github.com/aeternity/ae-channel-service/tree/early-reestablish

File: README.md [hunk 1]
```diff
@@ -55,8 +55,17 @@ Start the samples at your prompt by doing
...
 ```
 
+or for testnet
+```bash
+AE_NODE_URL="wss://testnet.aeternity.io:443/channel" AE_NODE_NETWORK_ID="ae_uat" mix test
+```
+
 Scenarios executed can be found [here](apps/ae_socket_connector/test/ae_socket_connector_test.exs)
 
+> test are designed to execute on a quick mining node, thus the test will
```
Comment (Artur64) [inline README.md]:
```suggestion
> tests are designed to execute on a quick mining node, thus the test will 
```

File: README.md [hunk 2]
```diff
@@ -55,8 +55,17 @@ Start the samples at your prompt by doing
...
 
+or for testnet
+```bash
+AE_NODE_URL="wss://testnet.aeternity.io:443/channel" AE_NODE_NETWORK_ID="ae_uat" mix test
+```
+
 Scenarios executed can be found [here](apps/ae_socket_connector/test/ae_socket_connector_test.exs)
 
+> test are designed to execute on a quick mining node, thus the test will 
+fail when directed to testnet. This is however keps as referense.
```
Comment (Artur64) [inline README.md]:
```suggestion
fail when directed to testnet. This however is kept as reference.
```

File: README.md [hunk 3]
```diff
@@ -66,6 +75,15 @@ AE_NODE_NETWORK_ID="ae_channel_service_test" iex -S mix phx.server
...
 Now, point your browser to [http://localhost:4000/](http://localhost:4000/)
 
+## Run sample interactive web client (testnet)
+
+```bash
+AE_NODE_URL="wss://testnet.aeternity.io:443/channel" AE_NODE_NETWORK_ID="ae_uat" iex -S mix phx.server
+```
+
+point your browser to `http://localhost:4000/`
+> testnet is currently load balanced, you need to be persistent (try again) to get your channel up and running. Current recomended worksround is to host your own node.
```
Comment (Artur64) [inline README.md]:
```suggestion
> testnet is currently load balanced, you need to be persistent (try again) to get your channel up and running. Current recommended workaround is to host your own node.
```

File: apps/ae_socket_connector/lib/session_holder.ex [hunk 4]
```diff
@@ -115,34 +98,56 @@ defmodule SessionHolder do
...
+    case socketconnector_state.channel_id do
+      nil -> Logger.warn("Not persisting to disk, channel_id missing")
+      _ ->
+        case :dets.insert(file_ref, {socketconnector_state.channel_id, dets_state}) do
+          :ok ->
+            case :dets.sync(file_ref) do
+              :ok -> :ok
+                case socketconnector_state.fsm_id == nil do
+                  true ->
+                    Logger.warn("Persisted data not satisfing reestablish requirements, fsm_id: #{inspect socketconnector_state.fsm_id} channel_id #{inspect socketconnector_state.channel_id}")
```
Comment (Artur64) [inline apps/ae_socket_connector/lib/session_holder.ex]:
```suggestion
                    Logger.warn("Persisted data not satisfying reestablish requirements, fsm_id: #{inspect socketconnector_state.fsm_id} channel_id #{inspect socketconnector_state.channel_id}")
```

File: apps/ae_socket_connector/lib/session_holder.ex [hunk 5]
```diff
@@ -115,34 +98,56 @@ defmodule SessionHolder do
...
+                  false ->
+                    Logger.info("Persisted data SATISFING reestablish requirements, fsm_id: #{inspect socketconnector_state.fsm_id} channel_id #{inspect socketconnector_state.channel_id}")
```
Comment (Artur64) [inline apps/ae_socket_connector/lib/session_holder.ex]:
```suggestion
                    Logger.info("Persisted data SATISFYING reestablish requirements, fsm_id: #{inspect socketconnector_state.fsm_id} channel_id #{inspect socketconnector_state.channel_id}")
```

**Review by Artur64 — COMMENTED** [review]
Looks good overall, just some typos and suggestions to quick-fix them.

**Review by Artur64 — APPROVED** [review]

**Pull request merged**